wp-content/plugins/tsd-push-notification/inc/auto-send-subscription.php

```diff
@@ -0,0 +1,85 @@
+<?php
+// https://wordpress.stackexchange.com/a/88335/75147
+function tsd_pn_post_transition_post_status( $new_status, $old_status, $post ) {
+	if ( get_post_type( $post ) != 'post' ) {
+		return;
+	}
+
+	if ( $new_status == 'publish' && $old_status != 'publish' ) {
+		// The post is published
+		// See note below (for `save_post` action) as for why we need to do this.
+		update_post_meta( $post->ID, '_tsd_pn_sent', 'ready' );
+	}
+}
+add_action( 'transition_post_status', 'tsd_pn_post_transition_post_status', 10, 3 );
+
+function tsd_pn_post_save_post( $post_id, $post ) {
+	$tsd_pn_sent_post_meta = get_post_meta( $post_id, '_tsd_pn_sent', 'sent' );
+	if ( $tsd_pn_sent_post_meta && $tsd_pn_sent_post_meta == "ready" ) {
```

Comment:
@hesyifei Just making sure -- could there be any case in which `$tsd_pn_sent_post_meta` is undefined?

Reply:
I see, it is undefined when the post type is not equal to 'publish'.